### PR TITLE
[sdk-vpp#314] Use token ID key from api

### PR DESCRIPTION
**Pull request by Bolodya1997**
## Description
Starts SR-IOV token ID mechanism parameter from `api`.

Depends on https://github.com/networkservicemesh/api/pull/110.

## Issue link
Needed for networkservicemesh/sdk-vpp#314.

## How Has This Been Tested?

- [x] ~Added unit testing to cover~ Covered by existing unit testing
- [x] Tested manually
- [ ] Tested by integration testing
- [ ] Have not tested

## Types of changes

- [ ] Bug fix
- [ ] New functionallity
- [ ] Documentation
- [x] Refactoring
- [ ] CI

### CODE REVIEW

File: pkg/networkservice/common/resourcepool/server.go [hunk 1]
```diff
@@ -58,10 +60,12 @@ func NewServer(
 func (s *resourcePoolServer) Request(ctx context.Context, request *networkservice.NetworkServiceRequest) (*networkservice.Connection, error) {
 	logger := log.FromContext(ctx).WithField("resourcePoolServer", "Request")
 	conn := request.GetConnection()
-	tokenID, ok := conn.GetMechanism().GetParameters()[TokenIDKey]
+	tokenID, ok := conn.GetMechanism().GetParameters()[common.TokenIDKey]
 	if !ok {
-		logger.Infof("no token id present for client connection %v", conn)
-		return next.Server(ctx).Request(ctx, request)
+		return nil, errors.New("no token ID provided")
```
Comment (Bolodya1997) [inline pkg/networkservice/common/resourcepool/server.go:65]: @pperiyasamy 
What do you think of returning an error in such case and so using it with a `switchcase` server for the complex chains?
Example of usage with switchcase - https://github.com/networkservicemesh/sdk-vpp/blob/70d7b5353d50c9c81b5de0978d4d040cc22a4d18/pkg/networkservice/chains/xconnectns/server.go#L119-L124.

Comment (pperiyasamy) [reply pkg/networkservice/common/resourcepool/server.go:65]: yes @Bolodya1997, this is a perfect case for using switch case chain element, I will change this and adapt sdk-ovs to use switch case chain element to align for this change.

Comment (pperiyasamy) [reply pkg/networkservice/common/resourcepool/server.go:65]: @Bolodya1997 raised a review in `sdk-ovs` to address this with `switchcase ` https://github.com/networkservicemesh/sdk-ovs/pull/16, so please go ahead with returning an error here.

**Review by pperiyasamy — APPROVED** [review]

**Review by Mixaster995 — APPROVED** [review]

**Review by glazychev-art — APPROVED** [review]

**Comment by denis-tingaikin** [comment]
@Bolodya1997 Could you resolve conflicts?

**Comment by edwarnicke** [comment]
@Bolodya1997 How are things going looking at just doing the endpoint Switch approach in cmd-forwarder-vpp ?

**Comment by Bolodya1997** [comment]
> @Bolodya1997 How are things going looking at just doing the endpoint Switch approach in cmd-forwarder-vpp ?

Everything is great:
1. PR introducing `endpoint.Combine` - https://github.com/networkservicemesh/sdk/pull/1077
2. `endpoint.Combine` usage in `cmd-forwarder-vpp` - https://github.com/networkservicemesh/cmd-forwarder-vpp/blob/d50f7aab7dde0350e91509eb182784dc3b6a29fb/internal/xconnectns/server.go

We still need this PR to `sdk-sriov` to make everything work with `common.DeviceTokenIDKey` from `api`.

**Comment by edwarnicke** [comment]
>We still need this PR to sdk-sriov to make everything work with common.DeviceTokenIDKey from api.

@Bolodya1997 Why does sdk-vpp need to know anything about DeviceTokenID?

**Comment by Bolodya1997** [comment]
> > We still need this PR to sdk-sriov to make everything work with common.DeviceTokenIDKey from api.
> 
> @Bolodya1997 Why does sdk-vpp need to know anything about DeviceTokenID?

It is a PR to `sdk-sriov`, `sdk-vpp#314` is an issue ID.

**Comment by edwarnicke** [comment]
Doh... apologies :)

**Pull request merged**